File: tests/performance/component/echo.py
# Copyright 2024, Pulumi Corporation.

from typing import Any, Optional

import pulumi

class Echo(pulumi.CustomResource):
    def __init__(
        self,
        resource_name: str,
        echo: pulumi.Input[Any],
        opts: Optional[pulumi.ResourceOptions] = None,
    ):
        props = {
            "echo": echo,
        }
        super().__init__("testprovider:index:Echo", resource_name, props, opts)

    @property
    @pulumi.getter
    def echo(self) -> pulumi.Output[Any]:
        return pulumi.get(self, "echo")

File: tests/integration/multistack/nodejs/a/echo.ts
import * as pulumi from "@pulumi/pulumi";

export interface EchoArgs {
    echo?: pulumi.Input<string>;
}

export class Echo extends pulumi.CustomResource {
    public readonly echo?: pulumi.Output<string>;

    constructor(name: string, args: EchoArgs, opts?: pulumi.CustomResourceOptions) {
        super("testprovider:index:Echo", name, args, opts);
    }
}
